ci-operator/step-registry/cucushift/installer/check/azure/confidential/confidentialvm/cucushift-installer-check-azure-confidential-confidentialvm-commands.sh

```diff
@@ -0,0 +1,186 @@
+    master_osdisk_securityencryptiontype=$(yq-go r "${INSTALL_CONFIG}" 'controlPlane.platform.azure.osDisk.securityProfile.securityEncryptionType')
+fi
+
+if [[ "${ENABLE_CONFIDENTIALVM_COMPUTE}" == "true" ]]; then
+    worker_encryptionathost=$(yq-go r "${INSTALL_CONFIG}" 'compute[0].platform.azure.encryptionAtHost')
+    worker_security_type=$(yq-go r "${INSTALL_CONFIG}" 'compute[0].platform.azure.settings.securityType')
+    worker_secure_boot=$(yq-go r "${INSTALL_CONFIG}" 'compute[0].platform.azure.settings.confidentialVM.uefiSettings.secureBoot')
+    worker_vtpm=$(yq-go r "${INSTALL_CONFIG}" 'compute[0].platform.azure.settings.confidentialVM.uefiSettings.virtualizedTrustedPlatformModule')
+    worker_osdisk_securityencryptiontype=$(yq-go r "${INSTALL_CONFIG}" 'compute[0].platform.azure.osDisk.securityProfile.securityEncryptionType')
+fi
+
+master_nodes_list=$(oc get nodes --selector='node-role.kubernetes.io/master' --no-headers | awk '{print $1}')
+for node in ${master_nodes_list}; do
+    if vm_confidentialvm_check "${node}" "${master_encryptionathost}" "${master_security_type}" "${master_secure_boot}" "${master_vtpm}" "${master_osdisk_securityencryptiontype}"; then
+        echo -e "INFO: confidentailVM check on node ${node} passed!\n"
+    else
+        echo -e "ERROR: confidentailVM check on node ${node} failed!\n"
+        critical_check_result=1
+    fi   
+done
+
+worker_nodes_list=$(oc get nodes --selector='node-role.kubernetes.io/worker' --no-headers | awk '{print $1}')
+for node in ${worker_nodes_list}; do
+    if vm_confidentialvm_check "${node}" "${worker_encryptionathost}" "${worker_security_type}" "${worker_secure_boot}" "${worker_vtpm}" "${worker_osdisk_securityencryptiontype}"; then
+        echo -e "INFO: confidentailVM check on node ${node} passed!\n"
+    else
+        echo -e "ERROR: confidentailVM check on node ${node} failed!\n"
+        critical_check_result=1
+    fi
+done
+
+# gen2 image definition check
+echo -e "\nGen2 image definition check..."
+ocp_minor_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f2)
+expected_image_security_type="${master_security_type:-$worker_security_type}"
+expected_image_security_type="${expected_image_security_type/VM/Vm}"
+if (( ${ocp_minor_version} > 17 )); then
+    expected_image_security_type="${expected_image_security_type}Supported"
+fi
+image_def_security_type=$(az sig image-definition show --gallery-image-definition ${INFRA_ID}-gen2 --gallery-name gallery_${INFRA_ID//-/_} -g ${RESOURCE_GROUP} --query "features[?name=='SecurityType'].value" -otsv)
+if [[ "${image_def_security_type}" == "${expected_image_security_type}" ]]; then
+    echo "INFO: Gen2 image defintion has expected feature setting!"
+else
+    echo "ERROR: Gen2 image definition has unexpected feature setting, expected feature: ${expected_image_security_type}, actual vaule: ${image_def_security_type}."
+    critical_check_result=1
+fi
+
+exit ${critical_check_result}
```

Comment:
The step is similar to `cucushift-installer-check-azure-confidential-trustedlaunch`, the main difference is one more check point for `osDisk.securityProfile.securityEncryptionType`, I am thinking if we can merge `cucushift-installer-check-azure-confidential-trustedlaunch` and this step as `cucushift-installer-check-azure-confidential`.